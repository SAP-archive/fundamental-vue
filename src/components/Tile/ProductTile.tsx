import { Slot, Component, Prop, Base } from '@/core';

interface Props {
    url?: string | null;
    title?: string | null;
    isButton?: boolean;
}

@Component('ProductTile')
@Slot('content', 'Product Content')
export class ProductTile extends Base<Props> {
    @Prop('image url', { type: String, default: null, required: true })
    public url!: string | null;

    @Prop({ type: String, default: null })
    public title!: string | null;

    @Prop('render product tile as a button', { type: Boolean, default: false })
    public isButton!: boolean;

    @Prop('disable product tile', { type: Boolean, default: false })
    public disabled!: boolean;

    private get style() {
        const url = this.url;
        if (url == null) {
            return {};
        }
        return {
            'background-image': `url(${this.url})`,
        };
    }

    public render() {
        const content = this.$slots.content;
        const title = this.title;
        const isButton = this.isButton;
        const disabled = this.disabled;

        return (
            <div class='fd-product-tile' role={isButton ? 'button' : null} aria-disabled={disabled}>
                <div class='fd-product-tile__media' style={this.style} />
                <div class='fd-product-tile__content'>
                    {title && <h2 class='fd-product-tile__title'>{title}</h2>}
                    {content}
                </div>
            </div>
        );
    }
}
