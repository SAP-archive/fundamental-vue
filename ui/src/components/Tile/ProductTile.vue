<template>
  <div class="fd-product-tile" :role="role" :aria-disabled="disabled">
    <div class="fd-product-tile__media" :style="style" />
    <div class="fd-product-tile__content">
      <h2 v-if="title != null" class="fd-product-tile__title">{{ title }}</h2>
      <slot name="content" />
    </div>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
export default Vue.extend({
  name: "FdProductTile",
  props: {
    url: { type: String, default: null },
    title: { type: String, default: null },
    isButton: { type: Boolean, default: null },
    disabled: { type: String, default: null }
  },
  computed: {
    style(): object {
      const url = this.url;
      if (url == null) {
        return {};
      }
      return {
        "background-image": `url(${this.url})`
      };
    },
    role(): string | null {
      return this.isButton ? "button" : null;
    }
  }
});
</script>
