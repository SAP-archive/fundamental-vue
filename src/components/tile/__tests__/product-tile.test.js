import { mount } from '@vue/test-utils'
import ProductTile from '../product-tile.vue'

describe('ProductTile', () => {
  const url = 'https://techne.yaas.io/images/product-thumbnail-wide.png'
  const title = 'Default Product Tile'

  const wrapper = mount(ProductTile, {
    propsData: {
      url,
      title
    }
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders correct props', () => {
    expect(wrapper.find('.fd-product-tile__title').text()).toEqual(title)
    expect(wrapper.find('.fd-product-tile__media').attributes('style')).toContain(
      `background-image: url(${url});`
    )
  })

  describe('As Button', () => {
    const wrapper = mount(ProductTile, {
      propsData: {
        url,
        title,
        isButton: true
      }
    })

    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('has role button', () => {
      expect(wrapper.attributes('role')).toEqual('button')
    })
  })
})
