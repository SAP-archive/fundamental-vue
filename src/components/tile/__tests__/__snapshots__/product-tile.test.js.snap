// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTile As Button renders correctly 1`] = `
<div
  class="fd-product-tile"
  role="button"
>
  <div
    class="fd-product-tile__media"
    style="background-image: url(https://techne.yaas.io/images/product-thumbnail-wide.png);"
  />
   
  <div
    class="fd-product-tile__content"
  >
    <h2
      class="fd-product-tile__title"
    >
      Default Product Tile
    </h2>
     
  </div>
</div>
`;

exports[`ProductTile renders correctly 1`] = `
<div
  class="fd-product-tile"
>
  <div
    class="fd-product-tile__media"
    style="background-image: url(https://techne.yaas.io/images/product-thumbnail-wide.png);"
  />
   
  <div
    class="fd-product-tile__content"
  >
    <h2
      class="fd-product-tile__title"
    >
      Default Product Tile
    </h2>
     
  </div>
</div>
`;
